log-replication/src/main/java/org/corfudb/logreplication/infrastructure/CorfuInterClusterReplicationServer.java

```diff
@@ -215,12 +219,18 @@ private void startServer() {
         Map<String, Object> opts = new Docopt(USAGE)
                 .withVersion(GitRepositoryState.getRepositoryState().describe)
                 .parse(args);
-
         printStartupMsg(opts);
         configureLogger(opts);
 
         log.info("Started with arguments: {}", opts);
-        this.siteManagerAdapter = constructSiteManagerAdapter();
+
+        if (opts.get("--site-manager-config") != null) {
+            String siteConfigPath = (String) opts.get("--site-manager-config");
+            this.siteManagerAdapter = new DefaultSiteManager(siteConfigPath);
```

Comment:
there is a path down the pipeline where the DefaultSiteManager would be created, from the specified --plugin (if that is the class you want to represent your adapter)